### PR TITLE
Fixes #38044 - Don't accept negative values for OS major version

### PR DESCRIPTION
**Pull request by girijaasoni**

### CODE REVIEW

File: app/models/operatingsystem.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Operatingsystem < ApplicationRecord
   accepts_nested_attributes_for :os_default_templates, :allow_destroy => true,
     :reject_if => :reject_empty_provisioning_template
 
-  validates :major, numericality: true, presence: { message: N_("Operating System version is required") }
+  validates :major, numericality: {:greater_than_or_equal_to => 0, :message => N_("Major version should be greater than or equal to 0") }, presence: { message: N_("Operating System version is required") }
```
Comment (Lennonka) [inline app/models/operatingsystem.rb]:
```suggestion
  validates :major, numericality: {:greater_than_or_equal_to => 0, :message => N_("Major version of the operating system must be greater than or equal to 0") }, presence: { message: N_("Operating System version is required") }
```
I'm not sure what the context for this text is, but I'd include the words "operating system". Also, it's a hard requirement, so "must".

Comment (girijaasoni) [reply app/models/operatingsystem.rb]: addressed

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM** works as expected

**Comment by stejskalleos** [comment]
Thanks @Lennonka @girijaasoni

**Pull request merged**